aws/resource_aws_neptune_parameter_group.go

```diff
@@ -0,0 +1,266 @@
+
+	return nil
+}
+
+func resourceAwsNeptuneParameterGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	d.Partial(true)
+
+	if d.HasChange("parameter") {
+		o, n := d.GetChange("parameter")
+		if o == nil {
+			o = new(schema.Set)
+		}
+		if n == nil {
+			n = new(schema.Set)
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		toRemove, err := expandNeptuneParameters(os.Difference(ns).List())
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[DEBUG] Parameters to remove: %#v", toRemove)
+
+		toAdd, err := expandNeptuneParameters(ns.Difference(os).List())
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[DEBUG] Parameters to add: %#v", toAdd)
+
+		// We can only modify 20 parameters at a time, so walk them until
+		// we've got them all.
+		maxParams := 20
+
+		for len(toRemove) > 0 {
+			paramsToModify := make([]*neptune.Parameter, 0)
+			if len(toRemove) <= maxParams {
+				paramsToModify, toRemove = toRemove[:], nil
+			} else {
+				paramsToModify, toRemove = toRemove[:maxParams], toRemove[maxParams:]
+			}
+			resetOpts := neptune.ResetDBParameterGroupInput{
+				DBParameterGroupName: aws.String(d.Get("name").(string)),
+				Parameters:           paramsToModify,
+			}
+
+			log.Printf("[DEBUG] Reset Neptune Parameter Group: %s", resetOpts)
+			err := resource.Retry(30*time.Second, func() *resource.RetryError {
+				_, err = conn.ResetDBParameterGroup(&resetOpts)
+				if err != nil {
+					if isAWSErr(err, "InvalidDBParameterGroupState", " has pending changes") {
+						return resource.RetryableError(err)
+					}
+					return resource.NonRetryableError(err)
+				}
+				return nil
+			})
+			if err != nil {
+				return fmt.Errorf("Error resetting Neptune Parameter Group: %s", err)
+			}
+		}
+
+		for len(toAdd) > 0 {
+			paramsToModify := make([]*neptune.Parameter, 0)
+			if len(toAdd) <= maxParams {
+				paramsToModify, toAdd = toAdd[:], nil
+			} else {
+				paramsToModify, toAdd = toAdd[:maxParams], toAdd[maxParams:]
+			}
+			modifyOpts := neptune.ModifyDBParameterGroupInput{
+				DBParameterGroupName: aws.String(d.Get("name").(string)),
+				Parameters:           paramsToModify,
+			}
+
+			log.Printf("[DEBUG] Modify Neptune Parameter Group: %s", modifyOpts)
+			_, err = conn.ModifyDBParameterGroup(&modifyOpts)
+			if err != nil {
+				return fmt.Errorf("Error modifying Neptune Parameter Group: %s", err)
+			}
+		}
+
+		d.SetPartial("parameter")
+	}
+
+	d.Partial(false)
+
+	return resourceAwsNeptuneParameterGroupRead(d, meta)
+}
+
+func resourceAwsNeptuneParameterGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	return resource.Retry(3*time.Minute, func() *resource.RetryError {
+		deleteOpts := neptune.DeleteDBParameterGroupInput{
+			DBParameterGroupName: aws.String(d.Id()),
+		}
+		_, err := conn.DeleteDBParameterGroup(&deleteOpts)
+		if err != nil {
+			awsErr, ok := err.(awserr.Error)
```

Comment:
Minor nitpick: we have an available helper function for SDK error handling and should prefer the SDK provided constants:

```go
if isAWSErr(err, neptune.ErrCodeDBParameterGroupNotFoundFault, "") {
  return nil
}
if isAWSErr(err, neptune.ErrCodeInvalidDBParameterGroupStateFault, "") {
  return resource.RetryableError(err)
}
return resource.NonRetryableError(err)
```